app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostActivity.java

```diff
@@ -0,0 +1,555 @@
+            null,
+            null,
+            bookOnDiskItem -> {
+              select(bookOnDiskItem);
+              return Unit.INSTANCE;
+            });
+    bookDelegate.setSelectionMode(SelectionMode.MULTI);
+    booksAdapter = new BooksOnDiskAdapter(bookDelegate,
+        BookOnDiskDelegate.LanguageDelegate.INSTANCE
+    );
+
+    presenter.attachView(this);
+
+    presenter.loadBooks();
+    recyclerViewZimHost.setAdapter(booksAdapter);
+
+    serviceConnection = new ServiceConnection() {
+
+      @Override
+      public void onServiceConnected(ComponentName className, IBinder service) {
+        HotspotService.HotspotBinder binder = (HotspotService.HotspotBinder) service;
+        hotspotService = binder.getService();
+        hotspotService.registerCallBack(ZimHostActivity.this);
+      }
+
+      @Override
+      public void onServiceDisconnected(ComponentName arg0) {
+      }
+    };
+
+    serviceIntent = new Intent(this, HotspotService.class);
+
+    startServerButton.setOnClickListener(v -> {
+      //Get the path of ZIMs user has selected
+      if (!ServerUtils.isServerStarted) {
+        getSelectedBooksPath();
+        if (selectedBooksPath.size() > 0) {
+          startHotspotHelper();
+        } else {
+          Toast.makeText(ZimHostActivity.this, R.string.no_books_selected_toast_message,
+              Toast.LENGTH_SHORT).show();
+        }
+      } else {
+        startHotspotHelper();
+      }
+    });
+  }
+
+  private void startHotspotHelper() {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+      toggleHotspot();
+    } else {
+      //TO DO: show Dialog() + within that add check mobile Data check later.
+      //if (isMobileDataEnabled(context)) {
+      //  mobileDataDialog();
+      //} else {
+      if (ServerUtils.isServerStarted) {
+        createHotspotIntent(ACTION_STOP_SERVER);
+      } else {
+        startHotspotManuallyDialog();
+      }
+      //}
+    }
+  }
+
+  private void getSelectedBooksPath() {
+    BooksOnDiskListItem.BookOnDisk bookOnDisk;
+
+    for (BooksOnDiskListItem item : booksAdapter.getItems()) {
+      if (item.isSelected()) {
+        bookOnDisk = (BooksOnDiskListItem.BookOnDisk) item;
+        File file = bookOnDisk.getFile();
+        selectedBooksPath.add(file.getAbsolutePath());
+        Log.v(TAG, "ZIM PATH : " + file.getAbsolutePath());
+      }
+    }
+  }
+
+  private void select(@NonNull BooksOnDiskListItem.BookOnDisk bookOnDisk) {
+    ArrayList<BooksOnDiskListItem> booksList = new ArrayList<>();
+    for (BooksOnDiskListItem item : booksAdapter.getItems()) {
+      if (item.equals(bookOnDisk)) {
+        item.setSelected(!item.isSelected());
+      }
+      booksList.add(item);
+    }
+    booksAdapter.setItems(booksList);
+  }
+
+  @Override protected void onStart() {
+    super.onStart();
+    bindService();
+  }
+
+  @Override protected void onStop() {
+    super.onStop();
+    unbindService();
+  }
+
+  private void bindService() {
+
```

Comment:
unecessary space

Reply:
Fixed in 8300dea416556ec795630c5f22243c3d22087abc